### PR TITLE
Multiple hitsound code

### PR DESCRIPTION
**Pull request by int-72h**
Added a small bit of code which allows for multiple custom hit/killsounds.
These are controlled with a few new convars:

``tf_dingalingaling_resetonkill``: When using 2+ hitsounds, if the hitsounds should go back to the initial one on kill.

``tf_dingalingaling_hitsound_num``: Number of hitsounds used, should be located in ``tf/sound/ui/hitsound-n.wav`` where n is the number beginning at 1.

``tf_dingalingaling_killsound_num``: Ditto of above for killsounds.

``tf_dingalingaling_hitsound_random``: If set to 1, the order in which the hitsounds are played (if there is multiple) will be random.

``tf_dingalingaling_killsound_random``: Ditto of above for killsounds.

I've tested this- I believe it should work.

@Nopey  helped a lot with this, so I don't deserve all the credit.
The `buf` size allocated should be enough for <10,000,000 hitsounds and <1,000,000 killsounds so this should be enough for anyone. (24-17/18 of path= 6/7 digits).

### CODE REVIEW

File: src/game/client/tf/tf_hud_damageaccount_panel.cpp [hunk 1]
```diff
@@ -388,9 +394,11 @@ void CDamageAccountPanel::PlayHitSound( int iAmount, bool bKill )
 	}
 
 	EmitSound_t params;
-
+	char buf[24]; // 7/8 digs for hitsounds should be enough, right?
```
Comment (int-72h) [inline src/game/client/tf/tf_hud_damageaccount_panel.cpp:397]: its probably actually 6/7 but it doesn't really matter

**Comment by kyokyoknutt** [comment]
Question: why is it called `dingalingaling`?
Is it just following TF2 silliness or something you threw in?
It probably would be a tighter UX if it was just something like "multihitsound". Not dissing on `dingalingaling`, that's pretty funny, but it would be a struggle to have to type `dingalingaling` multiple times to get things set the way you want. Or I'm just being nitpicky idfk

**Comment by int-72h** [comment]
I've had to nuke my current fork due to me screwing up the build files, so I'll remake this PR.

**Comment by int-72h** [comment]
Ok, timezones are _weird_. Says I wrote this an hour ago?

**Pull request closed without merging**